src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v1/TransactionClosePaymentRetryQueueConsumer.kt

```diff
@@ -175,10 +175,14 @@ class TransactionClosePaymentRetryQueueConsumer(
                   Pair(null, null)
                 }
               // transaction can be refund only for HTTP status code 422 and error response
-              // description equals to "Node did not receive RPT yet"
+              // description equals to "Node did not receive RPT yet" OR HTTP status code 400
+              // and error response description equal to "Unacceptable outcome when token has
+              // expired"
               val refundTransaction =
-                statusCode == HttpStatus.UNPROCESSABLE_ENTITY &&
-                  errorDescription == NodeClient.NODE_DID_NOT_RECEIVE_RPT_YET_ERROR
+                (statusCode == HttpStatus.UNPROCESSABLE_ENTITY &&
+                  errorDescription == NodeClient.NODE_DID_NOT_RECEIVE_RPT_YET_ERROR) ||
+                  (statusCode == HttpStatus.BAD_REQUEST &&
+                    errorDescription == NodeClient.UNACCEPTABLE_OUTCOME_TOKEN_EXPIRED)
```

Comment:
what about making a refactoring here and expose this logic as a ClosePaymentErrorResponseException function? 
So the above code will become something as
```code
val refundable = if (exception is ClosePaymentErrorResponseException){
 exception.isNodoRedundableError()) //change this orrible name xD
} else {
 false
}
```

This will reduce this pipeline cognitive complexity

Reply:
This can be implemented also using an [extension function](https://kotlinlang.org/docs/extensions.html#extension-functions) in order to make it extendible for future cases (not only ClosePaymentErrorResponseException)

something like:

```code
fun Throwable.isTransactionRefundable(): Boolean {
    if (this is ClosePaymentErrorResponseException) {
        val statusCode = this.statusCode
        val errorDescription = this.errorResponse?.description
        return (statusCode == HttpStatus.UNPROCESSABLE_ENTITY &&
                errorDescription == NodeClient.NODE_DID_NOT_RECEIVE_RPT_YET_ERROR) ||
                (statusCode == HttpStatus.BAD_REQUEST &&
                        errorDescription == NodeClient.UNACCEPTABLE_OUTCOME_TOKEN_EXPIRED)
    }
    return false
}
```

Reply:
I don't like the idea of applying an extension function to Throwable for a use case that is this specific to our context, very few exceptions would benefit of this extension (only one at the moment); furthermore, we would be forced to check the type of the exception to implement the right behaviour, reducing the benefits of extension. Since we are fully in control of those exceptions and behaviours, we can either modify it (as suggested by the first proposal) or, if we don't want to tie the refund eligibility to a certain exception, providing an utility class/method to check if the transaction that generated ClosePaymentErrorResponseException (or possible future exceptions) is eligible for a refund.